src/private/MDMViewControllerTransitionCoordinator.m

```diff
@@ -18,18 +18,25 @@
 
 #import "MDMTransition.h"
 
-@interface MDMViewControllerTransitionContext : NSObject <MDMTransitionContext>
-@property(nonatomic, strong) id<UIViewControllerContextTransitioning> transitionContext;
-@property(nonatomic, strong) id<MDMTransition> transition;
+@class MDMViewControllerTransitionContextNode;
+
+@protocol MDMViewControllerTransitionContextNodeParent <NSObject>
+- (void)childNodeTransitionDidEnd:(MDMViewControllerTransitionContextNode *)childNode;
 @end
 
-@protocol MDMViewControllerTransitionContextDelegate <NSObject>
-- (void)transitionContextDidEnd:(MDMViewControllerTransitionContext *)context;
+@interface MDMViewControllerTransitionContextNode : NSObject <MDMTransitionContext, MDMViewControllerTransitionContextNodeParent>
```

Comment:
Typical parent/node design pattern here. The parent protocol only exposes APIs needed by the child to communicate upwards.